internal/foundationdb_status.go

```diff
@@ -79,3 +81,50 @@ func GetMinimumUptimeAndAddressMap(cluster *fdbv1beta2.FoundationDBCluster, stat
 
 	return minimumUptime, addressMap, nil
 }
+
+// RemoveWarningsInJSON removes any warning messages that might appear in the status output from the fdbcli and returns
+// the JSON output without the warning message.
+func RemoveWarningsInJSON(jsonString string) ([]byte, error) {
+	idx := strings.Index(jsonString, "{")
+	if idx == -1 {
+		return nil, fmt.Errorf("the JSON string doesn't contain a starting '{'")
+	}
+
+	return []byte(strings.TrimSpace(jsonString[idx:])), nil
+}
+
+// GetUnsupportedClients will return all clients that are not supporting the specified protocolVersion. Clients that
+// have the same machine address as a fdbserver process will be ignored. The assumption here is that no other processes
+// are co-located to the fdbserver process. We have to ignore those clients since FDB 7.1 added all fdbserver processes
+// as clients and there is no other way to distinguish them and this would result in the operator being stuck in this step.
+// See: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1468
+func GetUnsupportedClients(status *fdbv1beta2.FoundationDBStatus, protocolVersion string) ([]string, error) {
+	fdbserverProcesses := map[string]fdbv1beta2.None{}
+	for _, process := range status.Cluster.Processes {
+		fdbserverProcesses[process.Address.MachineAddress()] = fdbv1beta2.None{}
+	}
+
+	var unsupportedClients []string
+	for _, versionInfo := range status.Cluster.Clients.SupportedVersions {
+		if versionInfo.ProtocolVersion == "Unknown" {
+			continue
+		}
+
+		if versionInfo.ProtocolVersion != protocolVersion {
+			for _, client := range versionInfo.MaxProtocolClients {
```

Comment:
My understanding is that `max_protocol_clients` is a sample of the clients that have this version as their max supported protocol. Does that mean it's possible that all of the clients in the sample will be fdbserver processes, but we will have normal clients that are not included in the sample, leading to false negatives from this check?

Reply:
I checked a 7.1 cluster with a large number of clients, and confirmed that the number of items in `max_protocol_clients` is smaller than the `max_protocol_count`, indicating that some of the clients are omitted.

Reply:
We discussed this offline and that's an expected behaviour. The list only contains a sample for the clients. This means we can't proceed with this PR as we might miss some real clients.